client/trino-cli/src/test/java/io/trino/cli/TestClientRedirect.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.cli;
+
+import com.google.common.collect.ImmutableList;
+import io.airlift.json.JsonCodec;
+import io.trino.client.ClientException;
+import io.trino.client.ClientSession;
+import io.trino.client.ClientTypeSignature;
+import io.trino.client.Column;
+import io.trino.client.QueryResults;
+import io.trino.client.StatementStats;
+import io.trino.client.uri.PropertyName;
+import io.trino.client.uri.TrinoUri;
+import okhttp3.logging.HttpLoggingInterceptor;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import okhttp3.mockwebserver.RecordedRequest;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.sql.SQLException;
+import java.util.Base64;
+import java.util.Optional;
+import java.util.OptionalDouble;
+import java.util.Properties;
+
+import static com.google.common.io.ByteStreams.nullOutputStream;
+import static com.google.common.net.HttpHeaders.AUTHORIZATION;
+import static com.google.common.net.HttpHeaders.CONTENT_TYPE;
+import static com.google.common.net.HttpHeaders.LOCATION;
+import static io.airlift.json.JsonCodec.jsonCodec;
+import static io.trino.cli.ClientOptions.OutputFormat.CSV;
+import static io.trino.cli.TerminalUtils.getTerminal;
+import static io.trino.cli.TestQueryRunner.createClientSession;
+import static io.trino.client.ClientStandardTypes.BIGINT;
+import static java.nio.charset.StandardCharsets.US_ASCII;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_METHOD;
+
+@TestInstance(PER_METHOD)
+public class TestClientRedirect
+{
+    private static final JsonCodec<QueryResults> QUERY_RESULTS_CODEC = jsonCodec(QueryResults.class);
+    private static final String DUMMY_ACCESS_TOKEN = Base64.getEncoder().encodeToString("jwt-signing-key".getBytes(US_ASCII));
+    private MockWebServer server;
+
+    @BeforeEach
+    public void setup()
+            throws IOException
+    {
+        server = new MockWebServer();
+        server.start();
+    }
+
+    @AfterEach
+    public void teardown()
+            throws IOException
+    {
+        server.close();
+        server = null;
+    }
+
+    @Test
+    public void testDisableClientRedirectsDefault()
```

Comment:
This test still pass when I modify `addNetworkInterceptor()` to `addInterceptor()`.
I think the test has to be redirect to another host:port.
Maybe one MockWebServer for proxy and one for Trino Coordinator.

Reply:
Added two mock servers, one for proxy one for coordinator